overlord/snapstate/snapstate.go

```diff
@@ -122,14 +126,23 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		prev = checkAsserts
 	}
 
+	// run refresh hooks when updating existing snap, otherwise run install
+	runRefreshHooks := (snapst.IsInstalled() && !snapsup.Flags.Revert)
```

Comment:
This should be moved right above the condition that uses it below. Otherwise this is lost between code that is completely unrelated to it.